kratos/utilities/nurbs_utilities/snake_sbm_utilities.h

```diff
@@ -0,0 +1,176 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+#pragma once
+
+// Project includes
+#include "includes/model_part.h"
+#include "spatial_containers/bins_dynamic.h"
+
+namespace Kratos
+{
+    ///@name Kratos Classes
+    ///@{
+
+    /**
+     * @class SnakeSBMUtilities
+     * @brief Utility class for implementing Snake-based Surrogate Boundary Method (SBM).
+     * This class provides various functions to create and manipulate snake-like surrogate boundaries
+     * for finite element models in Kratos.
+     */
+    class KRATOS_API(IGA_APPLICATION) SnakeSBMUtilities
+    {
+    public:
+        /**
+         * @brief Creates the initial snake coordinates for 2D or 3D skin.
+         * @param iga_model_part Model part containing the IGA geometry.
+         * @param skin_model_part_inner_initial Model part for the initial inner skin boundary.
+         * @param skin_model_part_outer_initial Model part for the initial outer skin boundary.
+         * @param skin_model_part Model part where the snake coordinates will be stored.
+         * @param rEchoLevel Verbosity level for logging.
+         * @param knot_vector_u Knot vector in the U-direction.
+         * @param knot_vector_v Knot vector in the V-direction.
+         * @param mParameters Input parameters for the process.
```

Comment:
I appreciate the docs 👍.

You need to explain what you expect as parameters here though.